SYS-1550: Add Discogs UPC search

ztucker4 opened the pull request:
Implements [SYS-1550](https://uclalibrary.atlassian.net/browse/SYS-1550)

Adds a few functions to get release data from Discogs for a given UPC and return a list of dictionaries with Title, Artist, Publisher Number, and full JSON.

Differences from MusicBrainz script:
* Discogs requires authentication for search, so I've added a `user_token` to `api_keys.py` and can send this file directly.
* Discogs search results contain less release info than in MusicBrainz, so we have to make one API call to search, and then another to get full release details. These are handled by `get_discogs_ids_by_upc()` and `get_full_discogs_releases()` respectively. Example use:

```
discogs_ids = get_discogs_ids_by_upc("018777260022")
discogs_releases = get_full_discogs_releases(discogs_ids)
parsed_results = parse_discogs_data(discogs_releases)
```

Includes tests (only for data parsing) with mocked versions of Release objects for simple cases of zero, one, and multiple results.

[SYS-1550]: https://uclalibrary.atlassian.net/browse/SYS-1550?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ztucker4 committed:
SYS-1550: Add Discogs UPC search

akohler reviewed (changes requested):
@ztucker4 The separation of data access and manipulation is good.  I'm able to use a Discogs token I already had to run routines from the main script.

For the tests, I think it's better to have a copy of some real data than to use mocks in this way.  I suggest capturing one "Release" in detail, dumping dict/json from `Release.data`, and storing that in a file which the tests then use.

ztucker4 committed:
restructure to support testing with real Discogs response

ztucker4 committed:
add test data file

ztucker4 commented:
Hi @akohler, this is ready for another look. I've added a sample data file with an actual Discogs release response which is loaded by the tests. There code is now updated to always use the `data` attribute from releases to create the output dictionary. This required using `.refresh()` to get full release data during the `get_full_discogs_releases()` step.

akohler reviewed (approved):
@ztucker4 The changes look good, thanks for storing real data for testing.  Good catch on `refresh()`, which gets barely mentioned in their documentation.

Merging now.
